File: backend/nightbiteApi/src/main/java/ni/edu/uam/nightbiteapi/repositories/PlayerRepository.java
package ni.edu.uam.nightbiteapi.repositories;

import ni.edu.uam.nightbiteapi.model.Player;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

/**
 * Repositorio encargado de gestionar las operaciones de base de datos
 * relacionadas con la ficha/personaje del repartidor.
 */
@Repository
public interface PlayerRepository extends JpaRepository<Player, Long> {

    Optional<Player> findByUserAccountId(Long userAccountId);

    boolean existsByUserAccountId(Long userAccountId);

    boolean existsByNickname(String nickname);
}

File: mobile/nightbiteApp/app/src/main/java/ni/edu/uam/nightbiteapp/data/repository/PlayerRepository.kt
package ni.edu.uam.nightbiteapp.data.repository

import ni.edu.uam.nightbiteapp.data.remote.ApiService
import ni.edu.uam.nightbiteapp.data.remote.RetrofitClient
import ni.edu.uam.nightbiteapp.data.remote.dto.PlayerRequest
import ni.edu.uam.nightbiteapp.data.remote.dto.PlayerResponse
import retrofit2.Response

/**
 * Repositorio encargado de gestionar las operaciones relacionadas
 * con la ficha/personaje del repartidor.
 */
class PlayerRepository(
    private val apiService: ApiService = RetrofitClient.apiService
) {

    suspend fun getPlayers(): Response<List<PlayerResponse>> {
        return apiService.getPlayers()
    }

    suspend fun getPlayerById(id: Long): Response<PlayerResponse> {
        return apiService.getPlayerById(id)
    }

    suspend fun getPlayerByUserAccountId(userAccountId: Long): Response<PlayerResponse> {
        return apiService.getPlayerByUserAccountId(userAccountId)
    }

    suspend fun createPlayer(playerRequest: PlayerRequest): Response<PlayerResponse> {
        return apiService.createPlayer(playerRequest)
    }
}